apps/cranio-provider/src/views/provider-cs-all-surgical.vue

```diff
@@ -4,223 +4,315 @@
 }
 
-function updateSurgicalComplications(value) {
-  const title = Object.keys(value);
-  surgicalComplicationsTitle.value = `${title} surgical complications`;
-  const total = value[Object.keys(value)];
-  setSurgicalComplications(total);
+function onSurgeryTypesChartClick(value: string) {
+  selectedSurgeryType.value = JSON.parse(value).dataPointName!;
+  updateComplicationsChart();
 }
 
-function setSurgicalInterventions() {
-  const types = Object.keys(surgicalInterventions.value);
-  let currentTotal = totalCases.value;
-  const data = types
-    .map((type, i) => {
-      const randomValue =
-        i === types.length - 1 ? currentTotal : randomInt(1, currentTotal)();
-      const row = [type, randomValue];
-      currentTotal -= randomValue;
-      return row;
-    })
-    .sort((current, next) => (current[1] < next[1] ? 1 : -1));
-  surgicalInterventions.value = Object.fromEntries(data);
+function updateComplicationsChart() {
+  const filteredData = complicationsChart.value?.dataPoints?.filter(
+    (row: IChartData) => {
+      return row.dataPointPrimaryCategory === selectedSurgeryType.value;
+    }
+  );
+
+  complicationsChartData.value = asKeyValuePairs(
+    filteredData,
+    "dataPointName",
+    "dataPointValue"
+  );
+
+  complicationsChartPalette.value = generateColorPalette(
+    Object.keys(complicationsChartData.value)
+  );
+
+  const sum: number = sumObjectValues(complicationsChartData.value);
+  hasComplicationsData.value = sum > 0;
 }
 
-function setAgeAtFirstSurgery() {
-  const ages = seq(0, 14, 2);
-  let currentTotal = totalCases.value;
-  const data = ages.map((age, i) => {
-    const randomValue =
-      i === ages.length - 1 ? currentTotal : randomInt(1, currentTotal)();
-    const row = { age: `${age}`, value: randomValue };
-    currentTotal -= randomValue;
-    return row;
-  });
-  ageAtFirstSurgery.value = data;
+function updateInterventionsChart() {
+  const filteredData = interventionsChart.value?.dataPoints!.filter(
+    (row: IChartData) => {
+      return row.dataPointPrimaryCategory === selectedDiagnosis.value;
+    }
+  );
+
+  interventionsChartData.value = asKeyValuePairs(
+    filteredData,
+    "dataPointValueLabel",
+    "dataPointValue"
+  );
+
+  const sum: number = sumObjectValues(interventionsChartData.value);
+  hasInterventionsData.value = sum > 0;
 }
 
-function onDiagnosisInput() {
-  setSurgicalInterventions();
-  setAgeAtFirstSurgery();
+function updateSurgeryAgeChart() {
+  surgeryAgeChartData.value = surgeryAgeChart.value?.dataPoints
+    ?.filter((row: IChartData) => {
+      return row.dataPointPrimaryCategory === selectedDiagnosis.value;
+    })
+    .sort((a: IChartData, b: IChartData) => {
+      return (a.dataPointTime as number) - (b.dataPointTime as number);
+    });
+
+  const chartAxis = generateAxisTickData(
+    surgeryAgeChartData.value,
+    "dataPointValue"
+  );
+
+  (surgeryAgeChart.value as ICharts).yAxisMaxValue = chartAxis.limit;
```

Comment:
I see a lot of 'as ...' like this one. I feel like this should not be necessary as the ref is typed. Am I missing something?